hw/ip/otbn/util/check_loop.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import sys
+from typing import Dict, List, Tuple
+
+from shared.check import CheckResult
+from shared.decode import OTBNProgram, decode_elf
+from shared.insn_yaml import Insn
+
+
+class CodeSection:
+    '''A continuous part of a program's code (represented as a PC range).
+
+    The code section is considered to include both the start and end PC.
+    '''
+    def __init__(self, start: int, end: int):
+        self.start = start
+        self.end = end
+
+    def __contains__(self, pc):
+        return self.start <= pc and pc <= self.end
+
+    def __repr__(self):
+        return '{:#x}-0x{:#x}'.format(self.start, self.end)
+
+
+def _get_pcs_for_mnemonic(program: OTBNProgram, mnem: str) -> List[int]:
+    '''Gets all PCs in the program at which the given instruction is present.'''
+    imem_start = program.get_pc_at_symbol('_imem_start')
+    imem_end = program.get_pc_at_symbol('_imem_end')
+    out = []
+    for pc in range(imem_start, imem_end, 4):
+        insn = program.get_insn(pc)
+        if insn.mnemonic == mnem:
+            out.append(pc)
```

Comment:
My comment about avoiding depending on `_imem_start` and `_imem_end` symbols goes here too. And the code gets so much shorter!
```python3
    for pc, (insn, _) in program.insns.items():
        if insn.mnemonic == mnem:
            out.append(pc)
```
or even (untested):
```python3
    return [pc for (pc, (insn, _)) in program.insns.items()
            if insn.mnemonic == mnem]
```
Look, ma! No loops!